Source/WebKit/WebProcess/Model/ModelProcessModelPlayer.cpp

```diff
@@ -77,38 +79,48 @@ bool ModelProcessModelPlayer::modelProcessEnabled() const
 
 // MARK: - Messages
 
-void ModelProcessModelPlayer::didCreateLayer(WebCore::LayerHostingContextIdentifier identifier)
+void ModelProcessModelPlayer::didCreateLayer(IPC::Connection& connection, WebCore::LayerHostingContextIdentifier identifier)
 {
     RELEASE_LOG(ModelElement, "%p - ModelProcessModelPlayer obtained new layerHostingContextIdentifier id=%" PRIu64, this, m_id.toUInt64());
+    MESSAGE_CHECK(modelProcessEnabled(), connection);
```

Comment:
What's the point of a MESSAGE_CHECK in the WebProcess? We usually use those in trusted processes, when validating IPC from untrusted processes.

Reply:
Technically this is not as useful as in privileged processes. We do print log if a message is received unexpectedly though, so this might be used to find bugs in trusted process (e.g. feature-flag-syncing issue), and I just decided to not change the behavior in this patch.

Reply:
@szewai But MESSAGE_CHECK doesn't seem to make sense in the WebProcess at all. MESSAGE_CHECK terminates the IPC connection normally. Not sure what MESSAGE_CHECK doesn't in the WebProcess. Seems we'd get more actionable crashes with a RELEASE_ASSERT. MESSAGE_CHECK is meant to kill the compromised remote process which doesn't really apply here.

Reply:
I think you are making a good point about WCP could be killed for bug; I will just change it to RELEASE_ASSERT

Reply:
Yes, we are going to drop `EnabledIf`.
`EnabledIf` currently overlaps with `EnabledBy` and `MESSAGE_CHECK`, and we are going to drop `EnabledIf` in favor of `EnabledBy` and `MESSAGE_CHECK`.
For feature flag protection (i.e. message endpoint enablement based on runtime feature flag), we should use `EnabledBy`; for others (validating parameters or status of message receiver object),  we should use `MESSAGE_CHECK` in message handler functions.
We are going to enforce `EnabledBy` annotation on new message endpoints soon. (We drop `EnabledIf` to avoid confusion on the requirement of feature flag protection).